Formula/d/decasify.rb

```diff
@@ -0,0 +1,40 @@
+class Decasify < Formula
+  desc "Utility for casting strings to title-case according to locale-aware style guides"
+  homepage "https://github.com/alerque/decasify"
+  url "https://github.com/alerque/decasify/releases/download/v0.6.0/decasify-0.6.0.tar.zst"
+  sha256 "01293c3300692bc4596ad8f91e9dbd0d0adb0bccf35a5ebac04047ce693d3c0b"
+  license "LGPL-3.0-only"
+
+  head do
+    url "https://github.com/alerque/decasify.git"
+
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
+    depends_on "libtool" => :build
+  end
+
+  depends_on "jq" => :build
+  depends_on "pkg-config" => :build
+  depends_on "rust" => :build
+
+  uses_from_macos "unzip" => :build
+  uses_from_macos "zlib"
+
+  def install
+    configure_args = %w[
+      --disable-silent-rules
+    ]
+
+    system "./bootstrap.sh" if build.head?
+    system "./configure", *configure_args, *std_configure_args
+    system "make"
+    system "make", "install"
+
+    (libexec/"bin").install bin/"decasify"
+    (bin/"decasify").write_env_script libexec/"bin/decasify", env
+  end
+
+  test do
+    assert_match "decasify #{version.to_s.match(/\d\.\d\.\d/)}", shell_output("#{bin}/decasify --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I added a test that actually exercises the expected behavior of the CLI. In particular it is using a non default language and making sure the output has the expected problematic Turkish dotted capital I:

```console
$ decasify -l tr -c title 'ben VE ivan'
Ben ve İvan
```

Of all the things likely not to work if anything at all was wrong, that is at the top of the list.